nft/src/main.sw

```diff
@@ -10,49 +10,56 @@ use abi::NFT;
 use data_structures::MetaData;
 use errors::{AccessError, ApprovalError, InitError, InputError};
 use events::{ApprovalEvent, BurnEvent, MintEvent, OperatorEvent, TransferEvent};
-use utils::{identities_equal, sender_identity};
+use utils::sender_identity;
 
 use std::{
-    address::Address,
     assert::require,
     chain::auth::{AuthError, msg_sender},
     context::{call_frames::{contract_id, msg_asset_id}, msg_amount, this_balance},
-    contract_id::ContractId,
     hash::sha256,
-    identity::Identity,
+    identity::*,
     logging::log,
     option::*,
     result::*,
     revert::revert,
     storage::StorageMap,
+    vec::Vec,
 };
 
 storage {
-    /// Determines if there is a whitelist to mint. Only set on the
-    /// initalization of the contract
+    /// Determines if there is an allow list to call the mint function
+    /// Only set on the initalization of the contract
     access_control: bool,
 
-    /// The identity that has permission to add identities to the whitelist
+    /// Stores the identity that has permission to add identities to the whitelist
+    /// Only set on the initalization of the contract
     access_control_address: Identity,
 
-    /// Stores the identities that are on the whitelist
+    /// Stores the identities that are on the allowed mint list
+    /// Map(Identity => bool)
     allowed_minters: StorageMap<Identity, bool>,
 
-    /// Maintains the number of tokens owned by each of the identities
+    /// Used of O(1) lookup of the number of tokens owned by each Identity
+    /// This increments or decrements when minting, transfer of ownership, and burning of tokens
+    /// Map(Identity => balance)
     balances: StorageMap<Identity, u64>,
 
-    /// The metadata for each token based on token id
+    /// Stores the Metadata for each token based on token id
+    /// Map(token_id => Metadata)
     meta_data: StorageMap<u64, Option<MetaData>>,
 
-    /// Stores a b256 hash of the (owner, operator) and stores whether the
+    /// Maps a b256 hash of the (owner, operator) identities and stores whether the
     /// operator is allowed to transfer ALL tokens on the owner's behalf 
+    /// Map(hash => approved)
     operator_approval: StorageMap<b256, bool>,
 
     /// TODO: Use a Vec here to support multiple ownerships
     /// Maintains the token ids owned by the specified identity
```

Comment:
This looks like it currently maps an `Identity` to the number of tokens it owns, rather than to the token_ids it owns. I'm guessing StorageVec is needed here.

Reply:
Right now since we do not have `StorageVec` it can only keep track of a single token, but it does store which token_ids are owned. In the case of transfer it currently just sets to 0 which is problematic if the `Identity` owns multiple tokens